recipes/pyoptsparse/meta.yaml

```diff
@@ -0,0 +1,82 @@
+{% set name = "pyoptsparse" %}
+{% set version = "2.6.2" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: "https://github.com/mdolab/pyoptsparse/archive/refs/tags/v{{ version }}.tar.gz"
+  sha256: 4f28eefa4eb5724f3d0a2b6118c45aedee24a7dea72a6672a49feb77220ecf32
+  patches:
+    - patch-IPOPT-setup-for-conda.patch
+
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}
+
+  host:
+    - python
+    - pip
+    - numpy
+    - swig
+    - ipopt 3.13.*
+    - libblas
+    - liblapack
+
+  run:
+    - python
+    - numpy
+    - sqlitedict >=1.6
+    - scipy >1.2
+    - mdolab-baseclasses >=1.3.1
+
+  run_constrained:
+    - ipopt 3.13.*
```

Comment:
This isn't needed as it will come from the run_exports of `ipopt`
```suggestion
```

Reply:
The intent was to make `ipopt` _optional_. However, when I looked at the rendered `meta.yaml` on commit c209e01f68817f83a1f9cc21448d856a337d42fd, it lists `ipopt` as a run requirement. I assume this is because the `run_exports` of the `ipopt` recipe add it to the run requirements.

I've pushed a change where I `ignore_run_exports` of `ipopt`, and then `pin_compatible(ipopt, ...)` in `run_constrained`. The resulting package does not list `ipopt` as a run requirement.

Is this totally wacky?

Reply:
It's better to have `ipopt` as a hard requirement instead of optional. There's no messaging that you can install these optional dependencies and the bot which sends pinning update PRs may not recognize these optional deps.

Reply:
Just FYI I am okay with putting `ipopt` as a hard dependency.